flybywiresim/aircraft
# feat(a380x/fcu): Fix visual issues with FCU light test and missing labels

Fixes #9091

## Summary of Changes


* Update display values on the FCU for the light test to match the reference image
* Change decimal point width in FCU font to match font and display in reference. This fixes the overlapping text issue in the speed display. This replaces an older PR #9129
* Move the FPA label to the left of the V/S label as per reference
* Add TRUE label. The label is not yet connected to TRUE mode but appears in the light test. Will be connected in a separate PR
* Move the degree symbol closer to the speed value to match the reference
* Changed first digit of FPA/VS to include different test digit (includes a +- symbol within the 8)
* Change FPA to not use the left most +- character. FPA +- will be aligned under the "F" character
* Add test pattern for EFIS CP labels

## Screenshots (if necessary)



Old image
![image](https://github.com/user-attachments/assets/e881a800-91da-4e33-9287-58b5201a0638)
![image](https://github.com/user-attachments/assets/fe8ed724-0dc0-41ba-b0c0-091930fa0434)

Updated Images
![image](https://github.com/user-attachments/assets/d697fc91-9952-48c4-bc64-8f1730644f97)
![FlightSimulator_2024-11-23-10-41-55-260](https://github.com/user-attachments/assets/b0f280c6-f7f7-44e5-be97-dfeba2adbc94)

FPA Symbol
![image](https://github.com/user-attachments/assets/308d6888-1f63-4cea-905b-90e21ec5723a)

## References


https://www.flickr.com/photos/75877584@N05/49491034416/in/photolist-2ipmuhQ-siX6c1-syfays-sAn7SW/

![image](https://imgproc.airliners.net/photos/airliners/3/3/8/2736833.jpg?v=v40)
https://www.airliners.net/photo/Lufthansa/Airbus-A380-841/2736833/L?qsp=eJwtjEEKwjAQRa8is3ZRKYh0Vz2ALrzAMPloaDVhMqCh9O5Og7vH%2B5%2B3kKS34Wv3mkEDFbDKk/aUWflVaFhoQv0kDc409qdud0ky5Wj%2BKUntXN0HNowiyIbw91cN0G1CkZZ7eP7gAL01pv7oPsSSZ24NGMeZ1vUHSF4wZA%3D%3D

![image](https://github.com/user-attachments/assets/47e4ae8e-973b-4650-a654-523f206f2797)




## Additional context



Discord username (if different from GitHub): Heclak

## Testing instructions


1. Set ANN LT sw to TEST. Check indication on FCU matches references and updated screenshots.
2. Switch modes on the FCU to check that labels are displaying/hiding correctly outside of TEST mode. Change numbers on FCU.
3. Check that preselect baro setting functions correctly. Preselect should timeout after a short time when in STD mode. 


## How to download the PR for QA

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, find and click on the **PR Build** tab
1. Click on either **flybywire-aircraft-a320-neo**, **flybywire-aircraft-a380-842 (4K)** or **flybywire-aircraft-a380-842 (8K)** download link at the bottom of the page
